Source/CurrencyCloud/Client.cs

```diff
@@ -1564,24 +1749,51 @@ private static Response CreateResponse(HttpStatusCode statusCode, HttpResponseHe
 
         private static async Task<List<Error>> CreateErrors(HttpContent content)
         {
-            string errorString = await content.ReadAsStringAsync();
-
-            JObject errorObject = JObject.Parse(errorString);
+            var errorString = await content.ReadAsStringAsync();
+            var errorObject = JObject.Parse(errorString);
 
-            var errors = from JProperty error in errorObject["error_messages"] 
-                select new Error(error.Name, 
-                    error.Value is JArray ? (from errorMessage in error.Value 
-                            select new Error.ErrorMessage(errorMessage["code"].Value<string>(),
-                                errorMessage["message"].Value<string>(),
-                                (from JProperty param in errorMessage["params"]
-                                    select new KeyValuePair<string, string>(param.Name, param.Value.ToString()))
-                                .ToDictionary(x => x.Key, x => x.Value)))
-                        .ToList() : new List<Error.ErrorMessage>(){new Error.ErrorMessage(error.Value["code"].Value<string>(), 
-                            error.Value["message"].Value<string>(), (from JProperty param in error.Value["params"]
-                                select new KeyValuePair<string, string>(param.Name, param.Value.ToString()))
-                            .ToDictionary(x => x.Key, x => x.Value))}
-                );
-            return errors.ToList();
+            try
```

Comment:
can this be done once without a need to parse it again in catch ? 
just tried to improve it with chatgpt, maybe it will work?

private static async Task<List<Error>> CreateErrors(HttpContent content)
{
    var errorString = await content.ReadAsStringAsync();
    var errorObject = JObject.Parse(errorString);

    var errors = from JProperty error in errorObject["error_messages"]
                 select new Error(
                     error.Name,
                     error.Value is JArray
                         ? (from errorMessage in error.Value
                            select new Error.ErrorMessage(
                                GetTokenValue(errorMessage, "code", "error_code"),
                                GetTokenValue(errorMessage, "message", "reason"),
                                GetParamsDictionary(errorMessage["params"])
                            )).ToList()
                         : new List<Error.ErrorMessage>
                         {
                             new Error.ErrorMessage(
                                 GetTokenValue(error.Value, "code", "error_code"),
                                 GetTokenValue(error.Value, "message", "reason"),
                                 GetParamsDictionary(error.Value["params"])
                             )
                         }
                 );

    return errors.ToList();
}

private static string GetTokenValue(JToken token, string primaryKey, string fallbackKey)
{
    return token[primaryKey]?.Value<string>() ?? token[fallbackKey]?.Value<string>();
}

private static Dictionary<string, string> GetParamsDictionary(JToken paramsToken)
{
    if (paramsToken == null)
    {
        return new Dictionary<string, string>();
    }

    return (from JProperty param in paramsToken
            select new KeyValuePair<string, string>(param.Name, param.Value.ToString()))
            .ToDictionary(x => x.Key, x => x.Value);
}